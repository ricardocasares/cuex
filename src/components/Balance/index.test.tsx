import { Balance } from "./index";
import renderer from "react-test-renderer";

describe("Balance", () => {
  test(`handles integers`, () =>
    expect(
      renderer
        .create(
          <Balance amount={120} currency={"USD"} overdraft={false}></Balance>
        )
        .toJSON()
    ).toMatchSnapshot());

  test(`handles floats`, () =>
    expect(
      renderer
        .create(
          <Balance amount={120.52} currency={"USD"} overdraft={false}></Balance>
        )
        .toJSON()
    ).toMatchSnapshot());

  test(`handles expressions`, () =>
    expect(
      renderer
        .create(
          <Balance amount={120.52} currency={"USD"} overdraft={3 > 2}></Balance>
        )
        .toJSON()
    ).toMatchSnapshot());

  test(`renders styles when overdraft`, () =>
    expect(
      renderer
        .create(
          <Balance amount={120.52} currency={"USD"} overdraft={true}></Balance>
        )
        .toJSON()
    ).toMatchSnapshot());
});
