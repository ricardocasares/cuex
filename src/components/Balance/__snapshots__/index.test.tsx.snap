// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance handles expressions 1`] = `
.emotion-0 {
  font-size: 14px;
  color: #fd1f8f;
}

<p
  className="emotion-0 emotion-1"
>
  Balance: 
  USD
  120.52
</p>
`;

exports[`Balance handles floats 1`] = `
.emotion-0 {
  font-size: 14px;
  color: #777;
}

<p
  className="emotion-0 emotion-1"
>
  Balance: 
  USD
  120.52
</p>
`;

exports[`Balance handles integers 1`] = `
.emotion-0 {
  font-size: 14px;
  color: #777;
}

<p
  className="emotion-0 emotion-1"
>
  Balance: 
  USD
  120
</p>
`;

exports[`Balance renders styles when overdraft 1`] = `
.emotion-0 {
  font-size: 14px;
  color: #fd1f8f;
}

<p
  className="emotion-0 emotion-1"
>
  Balance: 
  USD
  120.52
</p>
`;
